roles/cdi/tasks/version.yml

```diff
@@ -0,0 +1,15 @@
+---
+# 'version_compare' changes to 'version' in ansible-2.5
+- name: "cdi_release_tag >= 1.0 only works with >= kubevirt_version 0.7.0-alpha.4.  Setting cdi_release_tag to 0.5"
+  set_fact: cdi_release_tag="0.5"
+  when: "{{ '1.0' is version_compare(cdi_release_tag, '<=') }}
```

Comment:
1. `cdi_release_tag` >= 1.0   require `kubevirt_version` >= v0.7.0-alpha.4  (I agree with this condition)
2. `cdi_release_tag` < 1.0  can work with any `kubevirt_version` currently (my assumption)

even my assumption is wrong.
let's say `cdi_release_tag` == 1.0 and `kubevirt_version` == v0.7.1 in this case, then trigger `set_fact: cdi_release_tag="0.5"`, but why is this necessary ? 

maybe need to change to   `cdi_release_tag` < 1.0  && `kubevirt_version` > 0.7.0-alpha.4,  then downgrade `kubevirt_version` ?  (although I don't think it's necessary)

Reply:
Here's a matrix of the scenarios:
@aglitke can you confirm?

| | greater than or equal to 1.0 | less than 1.0 |
| --- | :---: | :---: |
| greater than or equal to 0.7.4-alpha.4 | X | |
| less than 0.7.4-alpha.4 | | X |

I need to set values if CDI is too new and KubeVirt is old and if CDI is too old and KubeVirt is new, otherwise the value is untouched.

Here, CDI is too new and KubeVirt old.  So I capture ```cdi_release_tag``` is greater than or equal to 1 and ```kubevirt_version``` less than 0.7.0-alpha.4.
```1.0 <= cdi_release_tag and 0.7.0-alpha.4 > kubevirt_version```
```yaml
"{{ '1.0' is version_compare(cdi_release_tag, '<=') }}
and {{ '0.7.0-alpha.4' is version_compare(kubevirt_version, '>') }}"
```
Results from ```kubevirt_version: "0.7.0-alpha.3``` and ```cdi_release_tag: "1.2"```:
```
...
TASK [idk : debug] **************************************************************************************************************************************************************************************************************************Friday 29 June 2018  08:38:39 -0400 (0:00:00.017)       0:00:00.107 ***********
ok: [localhost] => {}

MSG:

Using cdi_release_tag: 0.5
```

Next, CDI is too old and KubeVirt is new.  So I capture ```cdi_release_tag``` is less than 1 and ```kubevirt_version``` greater than or equal to 0.7.0-alpha.4.
```1.0 > cdi_release_tag and 0.7.0-alpha.4 <= kubevirt_version```
```yaml
"{{ '1.0' is version_compare(cdi_release_tag, '>') }}
and {{ '0.7.0-alpha.4' is version_compare(kubevirt_version, '<=') }}"
```
The results from running ```cdi_release_tag: 0.8``` and ```kubevirt_version: 0.7.0-alpha.4``` are:
```
TASK [idk : debug] **************************************************************************************************************************************************************************************************************************Friday 29 June 2018  08:50:44 -0400 (0:00:00.023)       0:00:00.099 ***********
ok: [localhost] => {}

MSG:

Using cdi_release_tag: 1.0
```
And finally, the results from running ```cdi_release_tag: 1.0``` and ```kubevirt_version: 0.7.0-alpha.4``` are that both tasks get skipped and ```cdi_release_tag``` is 1.0:
```
TASK [idk : cdi_release_tag >= 1.0 only works with >= kubevirt_version 0.7.0-alpha.4.  Setting cdi_release_tag to 0.5] **********************************************************************************************************************
Friday 29 June 2018  08:51:34 -0400 (0:00:00.056)       0:00:00.056 ***********
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: {{ '1.0' is version_compare(cdi_release_tag, '<=') }} and {{ '0.7.0-alpha.4' is version_compare(kubevirt_version, '>') }}

skipping: [localhost]

TASK [idk : kubevirt_version 0.7.0-alpha.4 requires cdi_release_tag >= 1.0.  Setting cdi_release_tag to 1.0] ********************************************************************************************************************************
Friday 29 June 2018  08:51:34 -0400 (0:00:00.016)       0:00:00.072 ***********
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: {{ '1.0' is version_compare(cdi_release_tag, '>') }} and {{ '0.7.0-alpha.4' is version_compare(kubevirt_version, '<=') }}

skipping: [localhost]

TASK [idk : debug] **************************************************************************************************************************************************************************************************************************
Friday 29 June 2018  08:51:34 -0400 (0:00:00.018)       0:00:00.091 ***********
ok: [localhost] => {}

MSG:

Using cdi_release_tag: 1.0
```